vllm/compilation/backends.py

```diff
@@ -164,80 +166,212 @@ def wrap_inductor(graph, example_inputs, additional_inductor_config):
+            else:
+                graph_for_symbolic_shape = graph
+
+            # TODO: Dynamo does not pass all dynamic shapes.
+            # Need to investigate why. It works now because all the dynamic
+            # shapes have the same value, and either of them can be used.
+            sym_shape_indices = [
+                i for i, x in enumerate(args) if isinstance(x, torch.SymInt)
+            ]
+
+            return graph(*args)
 
         if first_run:
-            # the first compilation is for profiling, we directly run it
             first_run = False
             return graph_for_symbolic_shape(*args)
 
+        runtime_shapes: Tuple[int,
+                              ...] = tuple(args[i] for i in sym_shape_indices)
+
         if runtime_shapes not in runtime_shapes_to_compile_flags:
             # we haven't seen this shape before
             # query if we need to specialize for this shape
             # we only specialize for the first dimension.
             # TODO: investigate if any model needs to specialize
             # beyond the first dimension
+
             runtime_shapes_to_compile_flags[runtime_shapes] = runtime_shapes[
-                0] in sizes_to_specialize
+                0] in compilation_configs.compile_sizes or \
+                runtime_shapes[0] in compilation_configs.capture_sizes
 
         if not runtime_shapes_to_compile_flags[runtime_shapes]:
             # we don't need to specialize for this shape
             return graph_for_symbolic_shape(*args)
 
-        if runtime_shapes not in runtime_shapes_to_compiled_graph:
-            # we need to specialize for this shape, and we haven't compiled
-            # compile the graph for this shape
-            logger.info("Compiling a graph for shapes %s", runtime_shapes)
-            runtime_shapes_to_compiled_graph[runtime_shapes] = wrap_inductor(
-                graph, args, additional_inductor_config)
-
-        return runtime_shapes_to_compiled_graph[runtime_shapes](*args)
+        if runtime_shapes not in runtime_shapes_to_compiled_entry:
+            # the first time we see this shape, we need to compile the graph
+            if compilation_configs.use_inductor and runtime_shapes[
+                    0] in compilation_configs.compile_sizes:
+                if is_first_graph:
+                    logger.info("Compiling a graph for shape %s",
+                                runtime_shapes)
+                runnable = wrap_inductor(
+                    graph, args, compilation_configs.inductor_compile_config)
+            else:
+                runnable = graph
+            use_cudagraph = compilation_configs.use_cudagraph and \
+                runtime_shapes[0] in compilation_configs.capture_sizes # noqa
+            entry = Entry(
+                runnable=runnable,
+                use_cudagraph=use_cudagraph,
+                target_warmup_times=compilation_configs.cudagraph_warmup_times)
```

Comment:
Does entry need to save `target_warmup_times`? Or can we just always uyse `compilation_configs.cudagraph_warmup_times`?